src/js/Content/Features/Community/Badges/FBadgeSortAndFilter.js

```diff
@@ -175,67 +175,77 @@ export default class FBadgeSortAndFilter extends Feature {
         this._hasAllPagesLoaded = true;
 
         const sheetNode = document.querySelector(".badges_sheet");
+        let images = HTMLParser.getVariableFromDom("g_rgDelayedLoadImages", "object");
 
-        // let images = Viewport.getVariableFromDom("g_rgDelayedLoadImages", "object");
-
-        await this.context.eachBadgePage(dom => {
-            for (const node of dom.querySelectorAll(".badge_row")) {
-                sheetNode.append(node);
-            }
+        await this.context.eachBadgePage((dom, delayedLoadImages) => {
+            sheetNode.append(...dom.querySelector(".badges_sheet").children);
+            images = Object.assign(images, delayedLoadImages);
 
             this.context.triggerCallbacks();
-
-            // images = Object.assign(images, Viewport.getVariableFromDom("g_rgDelayedLoadImages", "object", dom));
         });
 
         for (const node of document.querySelectorAll(".profile_paging")) {
             node.style.display = "none";
         }
 
-        /*
-         * TODO this doesn't seem to work, can't figure out why right now. Lazy loader doesn't see updated object?
-         * ExtensionLayer.runInPageContext("function(){g_rgDelayedLoadImages = " + JSON.stringify(images) + ";}");
-         * resetLazyLoader();
-         */
+        Page.runInPageContext(images => {
+            const f = window.SteamFacade;
+
+            f.globalSet("g_rgDelayedLoadImages", images);
+
+            // Clear registered image lazy loader watchers
+            CScrollOffsetWatcher.sm_rgWatchers = []; // eslint-disable-line no-undef, camelcase
+
+            // Recreate image lazy loader watchers
+            for (const node of document.querySelectorAll("div[id^=image_group_scroll_badge_images_]")) {
+                f.loadImageGroupOnScroll(node.id, node.id.slice(19));
+            }
+        }, [images]);
     }
 
     _sortBadgeRows(activeText, nodeValueCallback) {
+        const sheetNode = document.querySelector(".badges_sheet");
+
+        // Remove script tags that'll get in the way of associating rows with their scroll elements
+        for (const script of sheetNode.querySelectorAll(":scope > script")) {
+            script.remove();
+        }
```

Comment:
Is this necessary or does it just remove the need to to do `previousElementSibling` twice?

Reply:
Yes because if there're multiple pages, the HTML will be sanitized and have script tags removed before inserting to the current page, resulting in a discrepancy between badges on the current page and those appended later.

We could use `DOMParser` instead to parse the HTML and preserve all script tags, but then these script tags will have to be moved along with the badge row (and its scroll element) while sorting.